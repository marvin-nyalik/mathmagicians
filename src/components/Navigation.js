import { NavLink } from 'react-router-dom';
import NavCSS from './Navigation.module.css';

const navigation = () => (
  <nav className={NavCSS.nav}>
    <h2>
      <NavLink className={NavCSS.logo} to="/">Math-Magicians</NavLink>
    </h2>
    <ul className={NavCSS.ul}>
      <li>
        <NavLink className={NavCSS.navlink} to="/">Home</NavLink>
      </li>
      <li>
        <NavLink className={NavCSS.navlink} to="/calculator">Calculate</NavLink>
      </li>
      <li>
        <NavLink className={NavCSS.navlink} to="/quote">Quote</NavLink>
      </li>
    </ul>
  </nav>
);

export default navigation;
