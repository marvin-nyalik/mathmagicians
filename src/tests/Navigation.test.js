import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';

test('renders Navigation component correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
