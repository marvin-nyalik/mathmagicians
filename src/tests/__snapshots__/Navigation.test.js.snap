// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navigation component correctly 1`] = `
<nav
  class="nav"
>
  <h2>
    <a
      aria-current="page"
      class="logo active"
      href="/"
    >
      Math-Magicians
    </a>
  </h2>
  <ul
    class="ul"
  >
    <li>
      <a
        aria-current="page"
        class="navlink active"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class="navlink"
        href="/calculator"
      >
        Calculate
      </a>
    </li>
    <li>
      <a
        class="navlink"
        href="/quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
